### DISCUSSION TITLE
End of message in ssmanager

### DISCUSSION THREAD

**Discussion by anatolykopyl** [shadowsocks/shadowsocks-rust#1598] [Q&A]
I'm trying to write a program that communicates with the ssmanager, but can't figure out how to decide when the response is complete. 
It would be convenient if the message was followed by an EOF or a newline. What is the recommended way of dealing with this?

**Comment by zonyitoo** [comment]
Well, according to https://github.com/shadowsocks/shadowsocks/wiki/Manage-Multiple-Users , the protocol was designed to run on UDP. So the whole response will be put into a single datagram.

I am totally Ok to have a new stream based protocol, just like DNS over TCP/TLS/...
